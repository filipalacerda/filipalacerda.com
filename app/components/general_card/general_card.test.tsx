import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import GeneralCard from ".";

describe(GeneralCard.name, () => {
  let container: HTMLElement;

  beforeEach(() => {
    container = render(
      <GeneralCard title="Blog post title" href="custom.com" />,
    ).container;
  });

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("renders the provided title", () => {
    expect(screen.getByText("Blog post title")).toBeInTheDocument();
  });

  it("renders an external link", () => {
    const link = screen.getByRole("link");

    expect(link).toHaveAttribute("href", "custom.com");
    expect(link).toHaveAttribute("target", "_blank");
  });
});
