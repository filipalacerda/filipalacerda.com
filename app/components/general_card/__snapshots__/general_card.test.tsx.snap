// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralCard should match snapshot 1`] = `
<div>
  <section
    class="border rounded-sm p-4 mb-4 border-darkBlue"
  >
    <a
      class="text-green underline text-md font-light"
      href="custom.com"
      target="_blank"
    >
      Blog post title
    </a>
  </section>
</div>
`;
